src/server/coprocessor/endpoint.rs

```diff
@@ -676,7 +676,7 @@ impl<'a> SelectContext<'a> {
         let mut rows = vec![];
         let ids: Vec<i64> = {
             let cols = self.core.sel.get_index_info().get_columns();
-            cols.iter().map(|c| c.get_column_id()).collect()
+            cols.iter().filter(|c| !c.get_pk_handle()).map(|c| c.get_column_id()).collect()
```

Comment:
PKHancle column may be part of an index key, it can be removed only when it is the last column.